src/Compilers/CSharp/Portable/Binder/ForEachLoopBinder.cs

```diff
@@ -612,7 +612,8 @@ private BoundExpression UnwrapCollectionExpressionIfNullable(BoundExpression col
                     return BoundCall.Synthesized(
                         syntax: exprSyntax,
                         receiverOpt: collectionExpr,
-                        method: nullableValueGetter);
+                        method: nullableValueGetter,
+                        binder: this);
```

Comment:
This is the actual "bug". The rest are not in initial binding, and so don't need to specify a binder because the factory will never see them.